server/komodo-rest/src/main/java/org/komodo/rest/relational/KomodoRestUriBuilder.java

```diff
@@ -74,20 +166,20 @@ public URI buildVdbUri(final LinkType linkType, final String vdbName) {
     /**
      * @param linkType
      *        the type of URI being created (cannot be <code>null</code>)
-     * @param parentVdb
-     *        the parent VDB name (cannot be empty)
-     * @param modelName
-     *        the Vdb model name
+     * @param settings
+     *        configuration settings for this uri
      * @return the VDB model URI for the specified VDB (never <code>null</code>)
      */
-    public URI buildVdbModelUri(LinkType linkType, String parentVdb, String modelName) {
+    public URI buildVdbModelUri(LinkType linkType, final Properties settings) {
         ArgCheck.isNotNull(linkType, "linkType"); //$NON-NLS-1$
-        ArgCheck.isNotEmpty(parentVdb, "parentVdb"); //$NON-NLS-1$
-        ArgCheck.isNotEmpty(modelName, "modelName"); //$NON-NLS-1$
+        ArgCheck.isNotNull(settings, "settings"); //$NON-NLS-1$)
 
         URI result = null;
-        URI vdbUri = generateVdbUri(parentVdb);
-        URI myModelUri = generateVdbChildUri(parentVdb, LinkType.MODELS, modelName);
+        String parentName = vdbName(settings);
```

Comment:
I would move any local variable declarations inside the specific case statement block when they are only used within that one statement (like `vdbUri`). This comment applies to a few other methods as well.